sycl/include/sycl/accessor.hpp

```diff
@@ -1145,6 +1145,33 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
       : impl({}, detail::InitializedVal<AdjustedDim, range>::template get<0>(),
              detail::InitializedVal<AdjustedDim, range>::template get<0>()) {}
 
+  // implicit conversion between const / non-const types for read only accessors
+  template <typename DataT_, int Dimensions_, access::mode AccessMode_,
+            access::target AccessTarget_, access::placeholder IsPlaceholder_,
+            typename PropertyListT_>
+  accessor(const accessor<DataT_, Dimensions_, AccessMode_, AccessTarget_,
+                          IsPlaceholder_, PropertyListT_> &other,
+           std::enable_if_t<(AccessMode_ == access_mode::read) &&
+                            !(std::is_same<DataT_, DataT>::value) &&
+                            std::is_same<std::remove_const_t<DataT_>,
+                                         std::remove_const_t<DataT>>::value> * =
+               nullptr)
+      : impl(other.impl) {}
```

Comment:
Instead of defining these twice in the same class in difference `#ifdef __SYCL_DEVICE_ONLY__ ... #else` I think it would make more sense to define them publicly outside these and do something like:

```c++
template <...> accessor(...)
#ifdef __SYCL_DEVICE_ONLY__ 
 : impl(other.impl) {}
#else
: accessor<DataT, Dimensions_, AccessMode_, AccessTarget_, IsPlaceholder_,
                 PropertyListT_>(other.impl) {}
#endif
```

Reply:
thanks @steffenlarsen, that's better implementation indeed. Done.